### PR TITLE
feat(Core/Commands): deserter command offline & using targets & console support

### LINKED ISSUE #10826: Feature: deserter command should allow names as arguments including all
### Describe your feature request or suggestion in detail

Currently the `.deserter` command requires targeting a player to remove their deserters. If a player logs out, GMs can't work on their deserter auras.

### Describe a possible solution to your feature or suggestion in detail

The `.deserter` command should allow for an additional optional argument, to 
1) pass the name of a player. E.g. `.deserter bg add 3600 Hoggerhunter` or `.deserter dungeon remove Hoggerhunter` 
2) remove deserters from all players which last 30 minutes or less. E.g. `.deserter bg remove -1` (Useful after a crash)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gozzim**
## Changes Proposed:
- deserter command now works on offline targets
- deserter command now can take player arguments
-  Add `.deserter bg remove all` and `.deserter instance remove all` commands
- deserter command now works from the console

## Issues Addressed:

- Closes #10826

## Tests Performed:

- Ran the `.deserter` command in various variants

## How to Test the Changes:

1. Run the `.deserter` command using `add` and `remove` in all different variants from player & console on online & offline players

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Commands/cs_deserter.cpp [hunk 1]
```diff
@@ -139,43 +172,85 @@ class deserter_commandscript : public CommandScript
...
+    {
+        CharacterDatabase.Query("DELETE FROM character_aura WHERE spell = {} AND remainTime <= 1600000", isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+
+        std::shared_lock<std::shared_mutex> lock(*HashMapHolder<Player>::GetLock());
+        HashMapHolder<Player>::MapType const& onlinePlayerList = ObjectAccessor::GetPlayers();
+        for (HashMapHolder<Player>::MapType::const_iterator itr = onlinePlayerList.begin(); itr != onlinePlayerList.end(); ++itr)
+        {
+            Player* player = itr->second;
+            Aura* aura = player->GetAura(isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+            if (aura && aura->GetDuration() <= 1600000)
```
Comment (55Honey) [inline src/server/scripts/Commands/cs_deserter.cpp:212]: Can we please adjust this to cover the automated 30minutes deserter from RDF, too?
It should be `1800000`

Comment (Gozzim) [reply src/server/scripts/Commands/cs_deserter.cpp:212]: Oh damn. Thanks for pointing it out. I meant to do 30 minutes, so `1800000` but somehow messed up there :D

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by temperrr** [comment]
Tested, Works,

File: data/sql/updates/pending_db_world/rev_1653955272368406700.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--
+DELETE FROM `command` WHERE `name` IN ("deserter instance add", "deserter instance remove", "deserter instance remove all", "deserter bg add", "deserter bg remove", "deserter bg remove all");
+
+INSERT INTO `command` (`name`, `security`, `help`) VALUES
+("deserter instance add", 3, "Syntax: .deserter instance add $playerName $time \n\n Adds the instance deserter debuff to a player or your target with $time duration."),
+("deserter instance remove", 3, "Syntax: .deserter instance remove $playerName \n\n Removes the instance deserter debuff from a player or your target."),
+("deserter instance remove all", 3, "Syntax: .deserter instance remove all \n\n Removes the instance deserter debuff from all online and offline players."),
+("deserter bg add", 3, "Syntax: .deserter bg add $playerName $time \n\n Adds the bg deserter debuff to a player or your target with $time duration."),
+("deserter bg remove", 3, "Syntax: .deserter bg remove $playerName \n\n Removes the bg deserter debuff from a player or your target."),
+("deserter bg remove all", 3, "Syntax: .deserter bg remove all \n\n Removes the bg deserter debuff from all online and offline players.");
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1653955272368406700.sql:10]:
```suggestion
("deserter bg remove all", 3, "Syntax: .deserter bg remove all \n\n Removes the bg deserter debuff from all online and offline players.");

```

**Comment by Maelthyrr** [comment]
@Gozzim an issue here: https://github.com/azerothcore/azerothcore-wotlk/issues/11955

**Pull request merged**